src/Common/tests/System/Buffers/NativeOwnedMemory.cs

```diff
@@ -53,7 +53,11 @@ protected override bool IsRetained
 
         public override unsafe Span<byte> Span => new Span<byte>((void*)_ptr, _length);
 
-        public override unsafe MemoryHandle Pin() => new MemoryHandle(this, (void*)_ptr);
+        public override unsafe MemoryHandle Pin(int offset = 0)
```

Comment:
What happens when offset is negative? Is it important to handle that case in more robust way?

Reply:
Similar case - offset is more than the length of the block?

Reply:
The NativeOwnedMemory is implementing OwnedMemory for testing purposes only so it isn't crucial for this implementation to validate offset.

It makes sense that for an array backed OwnedMemory, we can't support negative offset, but for NativeOwnedMemory, should a negative offset be allowed?

Reply:
I do not see the difference between the two cases. Negative offset does not make sense to either, I think.